livekit/src/rtc_engine/rtc_session.rs

```diff
@@ -336,6 +341,10 @@ impl RtcSession {
     pub fn data_channel(&self, target: SignalTarget, kind: DataPacketKind) -> Option<DataChannel> {
         self.inner.data_channel(target, kind)
     }
+
+    pub async fn get_response(&self, request_id: u32) -> proto::RequestResponse {
+        self.inner.get_response(request_id).await
```

Comment:
I didn't see a timeout anywhere (maybe I missed it), would be good to have a timeout that rejects a request after a certain time in order for the request not to block indefinitely.